File: app/src/main/java/com/audiospotapplication/utils/DialogUtils.java
package com.audiospotapplication.utils;

import android.app.ProgressDialog;
import android.content.Context;

public class DialogUtils {

    private static ProgressDialog progressDialog;
    private static ProgressDialog playingProgressDialog;

    public static synchronized void showProgressDialog(Context context, String message) {
        if (progressDialog == null || !progressDialog.isShowing()) {
            progressDialog = new ProgressDialog(context);
            progressDialog.setMessage(message);
            progressDialog.setCancelable(false);
            progressDialog.show();
        }
    }

    public static synchronized Boolean isProgressShowing() {
        if (progressDialog == null)
            return false;
        return progressDialog.isShowing();
    }

    public static synchronized void dismissProgressDialog() {
        if (progressDialog != null && progressDialog.isShowing())
            progressDialog.dismiss();
    }

    public static synchronized void showDownloadingDialog(Context context, String message) {
        if (progressDialog == null || !progressDialog.isShowing()) {
            progressDialog = new ProgressDialog(context);
            progressDialog.setMessage(message);
            progressDialog.setCancelable(false);
            progressDialog.show();
        }
    }


    public static synchronized void showPlayingProgressDialog(Context context) {
        if (playingProgressDialog == null || !playingProgressDialog.isShowing()) {
            playingProgressDialog = new ProgressDialog(context);
            playingProgressDialog.setMessage("Playing ...");
            playingProgressDialog.setCancelable(false);
            playingProgressDialog.show();
        }
    }

    public static synchronized Boolean isPlayingProgressShowing() {
        if (playingProgressDialog == null)
            return false;
        return playingProgressDialog.isShowing();
    }

    public static synchronized void dismissPlayingProgressDialog() {
        if (playingProgressDialog != null && playingProgressDialog.isShowing())
            playingProgressDialog.dismiss();
    }
}

File: app/src/main/java/com/audiospotapplication/utils/DialogUtils.kt
package com.audiospotapplication.utils

import android.app.AlertDialog
import android.content.Context
import android.widget.TextView
import com.audiospotapplication.R

object DialogUtils {

    private var dialog: AlertDialog? = null

    @Synchronized
    fun showProgressDialog(
        context: Context?,
        message: String?
    ) {
        if (dialog == null || !dialog!!.isShowing) {
            val builder = AlertDialog.Builder(context)
            builder.setCancelable(false)
            builder.setView(R.layout.layout_loading_dialog)
            dialog = builder.create()
        }
        dialog!!.show()

        val tvLoadingMessage = dialog!!.findViewById<TextView>(R.id.loading)
        tvLoadingMessage.text = message
    }

    @get:Synchronized
    private val isProgressShowing: Boolean
        get() = if (dialog == null) false else dialog!!.isShowing

    @Synchronized
    fun dismissProgressDialog() {
        if (isProgressShowing) dialog!!.dismiss()
    }
}